test/functional/wallet_transactiontime_rescan.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test transaction time during old block rescanning
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal
+)
+
+
+class TransactionTimeRescanTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 3
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        self.log.info('Prepare nodes and wallet')
+
+        minernode = self.nodes[0] # node used to mine BTC and create transactions
+        usernode = self.nodes[1] # user node with correct time
+        restorenode = self.nodes[2] # node used to restore user wallet and check time determination in ComputeSmartTime (wallet.cpp)
+
+        # time constant
+        mocktime = 1525107225
+        ten_days = 10 * 24 * 60 * 60
+
+        # synchronize nodes and time
+        self.sync_all()
+        minernode.setmocktime(mocktime)
+        usernode.setmocktime(mocktime)
+        restorenode.setmocktime(mocktime)
+
+        # prepare miner wallet
+        minernode.createwallet(wallet_name='default')
+        miner_wallet = minernode.get_wallet_rpc('default')
+        m1 = miner_wallet.getnewaddress()
+
+        # prepare the user wallet with 3 public key
+        wo1 = 'bcrt1qqvs4h9g5kemwu8zdffv3c9tm3dm2qp9uzlxwzx'
+        wo2 = 'bcrt1qnw7peeq89hwrg674zjf4phzzluld798p667q7q'
+        wo3 = 'bcrt1qdr36v724uddf50jhnju3amzap2gmmt9tf4aac5'
+
+        usernode.createwallet(wallet_name='wo', disable_private_keys=True)
+        wo_wallet = usernode.get_wallet_rpc('wo')
+
+        wo_wallet.importpubkey(pubkey='02f8802962c2ca756d8961c719f75bdfff2b6dbb775788ed54a68bcb182c0ea1ef')
+        wo_wallet.importpubkey(pubkey='0330f4a96800a38fa0a49cde4b5f802e95a0c1044e5954389a8d00cecde01b842b')
+        wo_wallet.importpubkey(pubkey='02f5ccbee49546e0b5bf2e76422ed959e983dafaa94c8abafc9c8f8b99ef23d16a')
```

Comment:
The addresses and pubkeys don't need to be hardcoded. Have a look in other tests how this can be done (grep for "importpubkey", "deterministic", "pub_key", "priv_key", etc.)

Reply:
I've a question about this one. My first local version didn't use any hardcoded key and I use getaddressinfo and the pubkey field. I found my test a little bit hard to read, because I needed 4 wallets to realise it (and one, just to get a couple of address/pubkey). I find that by hardcoding address and pubkey, I could remove one wallet, and let the reader of the test be more focus on the 3 really important wallets.

Can you confirm me that I need to create a wallet to avoid hardcoded pubkey ? Or is there something I'm missing here.

If another wallet is required, I'll try to isolate it better from the 3 others than in my first attempt.

Reply:
I've find a way with getnewaddress() on the node. The test is rewrited with importaddress() instead of importpubkey(). It's simplier and demonstrated the same thing.
Done locally, not pushed yet.